src/web/apiClient.ts

```diff
@@ -22,17 +23,35 @@ export default class ApiClient {
             if (kestraInputUrl === undefined) {
                 vscode.window.showErrorMessage("Cannot get informations without proper Kestra URL.");
                 return "";
-            } else {
-                kestraUrl = kestraInputUrl;
-                // Store user URl
-                vscode.workspace.getConfiguration('kestra.api').update('url', kestraUrl, vscode.ConfigurationTarget.Global);
             }
+
+            finalUrl = this.formatApiUrl(kestraUrl);
+        }
+
+        // url was updated, we must save it to config
+        if (kestraUrl !== finalUrl) {
```

Comment:
This should be in the above if block to be sure that user cannot change kestraUrl in embedded vscode